dcpy/builds/plan.py

```diff
@@ -63,7 +63,9 @@ def dataset(self):
 class InputDatasetDefaults(BaseModel, use_enum_values=True):
     file_type: recipes.DatasetType | None = None
     preprocessor: DataPreprocessor | None = None
-    destination: InputDatasetDestination | None = None
+    destination: InputDatasetDestination = Field(
+        default=InputDatasetDestination.postgres, validate_default=True
```

Comment:
The fun dance of pydantic's `use_enum_values`
- If you just set default value to `InputDatasetDestination.postgres`, builds fail because conversion of enum to enum value happens at validation, and normally default values are not validated
- If you set default value to `InputDatasetDestination.postgres.value`, builds work just fine but `mypy` correctly sees that you're doing something that's not type safe

So here we are

Reply:
I think we might want to just write a custom `BaseModel` - just pydantic's basemodel, but with `extra="forbid"` and a custom deserializer so that we don't have to use `use_enum_values` and can still use the enums themselves within python while deserializing properly. I don't understand why there isn't a built-in implementation for that